ios/Plugin/CapacitorUpdater.swift

```diff
@@ -556,74 +557,173 @@ extension CustomError: LocalizedError {
         print("\(self.TAG) Current bundle set to: \((bundle ).isEmpty ? BundleInfo.ID_BUILTIN : bundle)")
     }
 
-    public func download(url: URL, version: String, sessionKey: String) throws -> BundleInfo {
-        let semaphore: DispatchSemaphore = DispatchSemaphore(value: 0)
-        let id: String = self.randomString(length: 10)
-        var checksum: String = ""
+    //Do a GET request on the url in order to extract the Content-Length header, ask only for the 10 first bytes in order to not download the full content.
```

Comment:
I believe that you could use a "head" request in order to get the content-length. As far as I know we do that [here](https://github.com/Cap-go/capgo/blob/5517026d4640659bd9020583225e01b4634d0047/supabase/functions/_backend/utils/s3.ts#L124)

Reply:
Apparently, the server doesn't support HEAD requests; at least, when I tried to do so, I got a 403 response from the server, so I had to work around with GET and the Range header.
![capgo head request](https://github.com/user-attachments/assets/a04fcc8f-4fe0-408b-ba42-5100ef071258)

Reply:
We do head request just here: https://github.com/Cap-go/capgo/blob/ebde1312a51feb6288ea5be750df71f3399cab46/supabase/functions/_backend/utils/s3.ts#L116
Just after getSignedUrl who is the command used to respond to the update, so it should work

Reply:
Please let me know if I'm missing anything. I attempted to use a `HEAD` request to retrieve the `Content-Length` header, but I kept receiving a `403` error, which suggests that the download URL may not support `HEAD` requests. To address this issue, in commit 90a09dd, I modified my `GET` request logic to include `Range: bytes=0-0`. This adjustment ensures that my `GET` request is basically the same as a `HEAD` request, effectively retrieving only the first byte and **providing the necessary headers**.
![403capgo](https://github.com/user-attachments/assets/bf3f1711-8ec2-4e74-aee8-e4dcad2a6f91)